blob/singularity_store.go

```diff
@@ -13,23 +13,31 @@ import (
 	"github.com/data-preservation-programs/singularity/client"
 	"github.com/data-preservation-programs/singularity/handler/dataset"
 	"github.com/data-preservation-programs/singularity/handler/datasource"
+	"github.com/data-preservation-programs/singularity/model"
 	"github.com/data-preservation-programs/singularity/service/epochutil"
 )
 
 const motionDatasetName = "MOTION_DATASET"
 const maxCarSize = "31.5GiB"
+const packThreshold = 1 << 34
 
 type SingularityStore struct {
 	local             *LocalStore
 	sourceID          uint32
 	singularityClient client.Client
+	toPack            chan uint64
+	closing           chan struct{}
+	closed            chan struct{}
 }
 
 func NewSingularityStore(dir string, singularityClient client.Client) *SingularityStore {
 	local := NewLocalStore(dir)
 	return &SingularityStore{
 		local:             local,
 		singularityClient: singularityClient,
+		toPack:            make(chan uint64, 16),
```

Comment:
how do you plan to handle the case that motion crashed/rebooted and the pending file IDs are lost